zerver/tests/test_markdown.py

```diff
@@ -576,18 +576,18 @@ def test_inline_image_thumbnail_url(self) -> None:
         realm = get_realm("zephyr")
         msg = "[foobar](/user_uploads/{realm_id}/50/w2G6ok9kr8AMCQCTNAUOFMln/IMG_0677.JPG)"
         msg = msg.format(realm_id=realm.id)
-        thumbnail_img = '<img data-src-fullsize="/thumbnail?url=user_uploads%2F{realm_id}%2F50%2Fw2G6ok9kr8AMCQCTNAUOFMln%2FIMG_0677.JPG&amp;size=full" src="/thumbnail?url=user_uploads%2F{realm_id}%2F50%2Fw2G6ok9kr8AMCQCTNAUOFMln%2FIMG_0677.JPG&amp;size=thumbnail"><'
+        thumbnail_img = '<img src="/user_uploads/3/50/w2G6ok9kr8AMCQCTNAUOFMln/IMG_0677.JPG">'
         thumbnail_img = thumbnail_img.format(realm_id=realm.id)
         converted = markdown_convert_wrapper(msg)
         self.assertIn(thumbnail_img, converted)
 
         msg = "https://www.google.com/images/srpr/logo4w.png"
-        thumbnail_img = '<img data-src-fullsize="/thumbnail?url=https%3A%2F%2Fwww.google.com%2Fimages%2Fsrpr%2Flogo4w.png&amp;size=full" src="/thumbnail?url=https%3A%2F%2Fwww.google.com%2Fimages%2Fsrpr%2Flogo4w.png&amp;size=thumbnail">'
+        thumbnail_img = '<img src="https://external-content.zulipcdn.net/external_content/56c362a24201593891955ff526b3b412c0f9fcd2/68747470733a2f2f7777772e676f6f676c652e636f6d2f696d616765732f737270722f6c6f676f34772e706e67">'
         converted = markdown_convert_wrapper(msg)
         self.assertIn(thumbnail_img, converted)
 
         msg = "www.google.com/images/srpr/logo4w.png"
-        thumbnail_img = '<img data-src-fullsize="/thumbnail?url=http%3A%2F%2Fwww.google.com%2Fimages%2Fsrpr%2Flogo4w.png&amp;size=full" src="/thumbnail?url=http%3A%2F%2Fwww.google.com%2Fimages%2Fsrpr%2Flogo4w.png&amp;size=thumbnail">'
+        thumbnail_img = '<img src="https://external-content.zulipcdn.net/external_content/7b6552b60c635e41e8f6daeb36d88afc4eabde79/687474703a2f2f7777772e676f6f676c652e636f6d2f696d616765732f737270722f6c6f676f34772e706e67">'
         converted = markdown_convert_wrapper(msg)
```

Comment:
This test is pretty explicitly supposed to be testing the thumbnailing case. We should make this test use `@override_settings(THUMBNAIL_IMAGES=True)` for some of these tests, rather than making it not test what it says it tests.

```
$ git grep THUMBNAIL_IMAGES
zerver/lib/markdown/__init__.py:            settings.THUMBNAIL_IMAGES
zerver/tests/test_markdown.py:        with self.settings(THUMBNAIL_IMAGES=False):
zerver/tests/test_markdown.py:    @override_settings(THUMBNAIL_IMAGES=False, EXTERNAL_URI_SCHEME="https://")
zerver/tests/test_markdown.py:    @override_settings(THUMBNAIL_IMAGES=False, EXTERNAL_URI_SCHEME="https://")
zerver/tests/test_markdown.py:    @override_settings(THUMBNAIL_IMAGES=False, EXTERNAL_URI_SCHEME="https://")
zerver/tests/test_markdown.py:    @override_settings(THUMBNAIL_IMAGES=False, EXTERNAL_URI_SCHEME="https://")
zproject/default_settings.py:THUMBNAIL_IMAGES = False
zproject/dev_settings.py:THUMBNAIL_IMAGES = True
zproject/test_extra_settings.py:THUMBNAIL_IMAGES = True
```